### ISSUE TITLE
Make native browser w/API usage

### ISSUE THREAD

**Issue by alfonsojon** [roblox-linux-wrapper/roblox-linux-wrapper#71]
Given that Roblox now has an API which can be used for authentication, the full Firefox build is overkill. A better solution is to authenticate with the API, then call wine directly from the browser without using the Roblox plugin. This would make the functionality similar to that of the old Roblox Browser.

API reference here:
https://github.com/matthewdean/roblox-web-apis

Demo in XUL (currently does not call RobloxPlayerBeta.exe, but works as a simple browser):
https://github.com/alfonsojon/roblox-browser

**Comment by alfonsojon** [comment]
This should be useful:
http://stackoverflow.com/questions/4315111/how-to-do-http-request-call-with-json-payload-from-command-line

**Comment by GLolol** [comment]
As far as I'm concerned, with Google Chrome recently dropping NPAPI plugin support, Roblox is already transitioning to plugin-less game starts for Chrome users. There was a news post about it a while ago, but I can't seem to find the link for it anymore.

Anyway, doing a site login from the wrapper itself would be nice, but I'm not sure how the internals would work. What engine should we use for the game browser? Any browser that uses a GUI will probably be written in a different language than Shell, and requires its own dependencies.

**Comment by alfonsojon** [comment]
@GLolol XULRunner seems like a good candidate - it's multi-platform, stable, easily deployed, and does not require any additional dependencies. The demo found on that GitHub repo linked above can actually run on Linux, OS X, and Windows, and is completely free of dependencies.

**Comment by GLolol** [comment]
How do I run the demo exactly?

**Comment by alfonsojon** [comment]
@GLolol You need xulrunner installed. Once installed, run `xulrunner /path/to/roblox-browser-xulrunner`

**Comment by GLolol** [comment]
Oh, neat. I was only able to run the demo using by calling the command on the `application.ini`. `iceweasel -app`/`firefox -app` seems to have replaced `xulrunner` entirely for me on Debian. 

```
gl@midnight:~/roblox-browser/roblox-browser$ iceweasel -app application.ini
```

**Comment by StripedFelines** [comment]
Are you still planning on doing this? @alfonsojon

**Comment by alfonsojon** [comment]
@StripedFelines See https://github.com/roblox-linux-wrapper/single-site-browser-vala

**Comment by alfonsojon** [comment]
Fixed with #163
